.github/workflows/build.yml

```diff
@@ -50,3 +53,61 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
         run: mvn -B verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar
+
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: build-artifacts
+          path: target/classes
+
+      - name: Set ref_name and commit_sha
+        id: set-vars
+        run: |
+          echo "::set-output name=ref_name::${{ github.ref_name }}"
+          echo "::set-output name=commit_sha::${{ github.sha }}"
+
+  deploy-to-gh-pages:
+    name: Deploy to GitHub Pages
+    runs-on: ubuntu-latest
+    needs: build
+    # Execute this stage only for main and tags
+    if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/')
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout gh-pages branch
+        uses: actions/checkout@v4
+        with:
+          ref: gh-pages
+          fetch-depth: 0
+
+      - name: Define target_data_dir
+        id: set-target-dir
+        run: echo "target_data_dir=data/${{ needs.build.outputs.ref_name }}" >> $GITHUB_ENV
+
+      - name: Remove old content
+        run: |
+          git clean -fd
+          git reset --hard
+          rm -rf "${{ env.target_data_dir }}"
+
+      - name: Download build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: build-artifacts
+          path: "${{ env.target_data_dir }}"
+
+      - name: Generate data/index.json
+        run: |
+          find data -name "index.json" | sort | while IFS= read -r indexJsonFile; do
+            relative_path="$(echo "${indexJsonFile}" | sed -E 's,^data/,,')"
+            jq --arg file "$relative_path" '.specification | {($file): {title: .title, version: .version, scmRevisionDate: .scmRevisionDate, scmRevisionNumber: .scmRevisionNumber}}' "$indexJsonFile"
```

Comment:
I understand this workflow step, but couldn't it be simplified by migrating the handling to the Java code during file generation?